### ISSUE TITLE
var-spacing: False-positive on regexp

### ISSUE THREAD

**Issue by charlesrocket** [ansible/ansible-lint#2208]
##### Summary

The pipe in regex triggers false-positive, i.e `'(sh|py)'`

```
var-spacing: Jinja2 variables and filters should have spaces before and after.
```

```
# example
loop: "{{ some_var | reject('match', '([a-zA-Z0-9-/\\s_\\.\\-\\(\\):])+\\.(lua|py)') | flatten(levels=1) | list }}"
```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible [core 2.12.4]
  config file = None
  configured module search path = ['/home/charlie/.ansible/plugins/modules', '/usr/local/share/py38-ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/charlie/.ansible/collections:/usr/local/share/py38-ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.13 (default, May 10 2022, 01:15:55) [Clang 11.0.1 (git@github.com:llvm/llvm-project.git llvmorg-11.0.1-0-g43ff75f2c
  jinja version = 3.0.1
  libyaml = True

ansible-lint 6.3.0 using ansible 2.12.4

```

- ansible installation method: OS package
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```console (paste below)

```

##### Desired Behavior

Regex should not trigger any errors.

##### Actual Behavior

The test is failing if `|` character within a regex variable is not separated with spaces.

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by max06** [comment]
@ssbarnea sorry for pinging...

```text
jinja[spacing]
Description: Jinja2 spacing could be improved: {{ parsed_checksum.parsed\[image.url | basename] }} -> {{ parsed_checksum.parsed\[image.url |basename] }}
```

The result is listed as a bad example in this PR. 
Running v6.5.1

**Comment by ssbarnea** [comment]
I know for sure that there are still few bugs with spacing, but I do not really understand the first one, especially as `\[` would be the expression invalid.

Try to make an example as short as possible for reproducing the bug and post it to https://github.com/ansible/ansible-lint/issues/2338 - we will use this to collect the bad formatting and in the folowing days to fix it.

**Comment by max06** [comment]
The `\[` might actually just be a display error in vscode. It's not coming from the sourcecode.

Shortened my two examples and added them.
